### PR TITLE
[Plots.Scatter] - Property to Attribute transformation logic done on scatterPlot now

### PR DESCRIPTION
**Pull request by bluong**
Used `generatePropertyToProjectors()` in order to transform properties to attributes

### CODE REVIEW

File: src/components/plots/scatterPlot.ts [hunk 1]
```diff
@@ -87,6 +81,26 @@ export module Plots {
...
     }
+
+    protected _generatePropertyToProjectors(): AttributeToProjector {
+      var propertyToProjectors = super._generatePropertyToProjectors();
+
+      var xProjector = propertyToProjectors["x"];
+      var yProjector = propertyToProjectors["y"];
+
+      var sizeProjector = propertyToProjectors[Scatter._SIZE_KEY];
+      // delete propertyToProjectors[Scatter._SIZE_KEY];
```
Comment (jtlan) [inline src/components/plots/scatterPlot.ts]: remove commented-out line.

File: src/components/plots/scatterPlot.ts [hunk 2]
```diff
@@ -87,6 +81,26 @@ export module Plots {
...
+
+      propertyToProjectors["transform"] = (datum: any, index: number, dataset: Dataset, plotMetadata: Plots.PlotMetadata) =>
+        "translate(" + xProjector(datum, index, dataset, plotMetadata) + "," + yProjector(datum, index, dataset, plotMetadata) + ")";
+
+      var symbolProjector = propertyToProjectors[Scatter._SYMBOL_KEY];
+      // delete propertyToProjectors[Scatter._SYMBOL_KEY];
```
Comment (jtlan) [inline src/components/plots/scatterPlot.ts]: remove commented-out line.

File: src/components/plots/scatterPlot.ts [hunk 3]
```diff
@@ -87,6 +81,26 @@ export module Plots {
 
       return Utils.Methods.intersectsBBox(xRange, yRange, translatedBbox);
     }
+
+    protected _generatePropertyToProjectors(): AttributeToProjector {
```
Comment (jtlan) [inline src/components/plots/scatterPlot.ts]: Maybe we want to name this method `_propertyProjectors()`?

File: src/components/plots/scatterPlot.ts [hunk 4]
```diff
@@ -87,6 +81,26 @@ export module Plots {
...
+      var propertyToProjectors = super._generatePropertyToProjectors();
+
+      var xProjector = propertyToProjectors["x"];
+      var yProjector = propertyToProjectors["y"];
```
Comment (jtlan) [inline src/components/plots/scatterPlot.ts]: Hmm, if we're just going to use the strings `"x"` and `"y"` here, should we remove the `static` keys from the superclass?

Alternatively, the `static` keys should be public, and we should use them here.

**Comment by jtlan** [comment]
[-1] This looks awesome, just some minor cleanup.

**Comment by bluong** [comment]
I'm going to do more and more cleanup as I dig into the API and thus, I think
> And then have a `_projectors()` method to combine the ones from properties with the ones from attributes?
is out of scope.  I will highly consider a much more concise implementation as time goes on

File: src/components/plots/scatterPlot.ts [hunk 5]
```diff
@@ -23,20 +23,14 @@ export module Plots {
                                            .delay(5));
       this.attr("opacity", 0.6);
       this.attr("fill", new Scales.Color().range()[0]);
+      this.size(6);
+      this.symbol(() => SymbolFactories.circle());
```
Comment (jtlan) [inline src/components/plots/scatterPlot.ts]: Does this create a new `SymbolFactory` each time it's invoked? If so, you might consider doing:
```Typescript
var circleFactory = SymbolFactories.circle();
this.symbol(() => circleFactory);
```

**Comment by jtlan** [comment]
[0] Minor question on efficiency of the default `symbol` `Accessor`.

**Pull request merged**